lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -497,25 +443,40 @@ def get_kv_requests(self, relation_id: int | None = None) -> List[KVRequest]:
                 )
         return kv_requests
 
-    def _credentials_issued_for_request(self, nonce: str, relation_id: int) -> bool:
-        """Return whether credentials have been issued for the request."""
-        relation = self.model.get_relation(self.relation_name, relation_id)
-        if not relation:
-            return False
+    def set_kv_data(
+        self,
+        relation: ops.Relation,
+        mount: str,
+        ca_certificate: str,
+        vault_url: str,
+        nonce: str,
+        secret_id: str,
```

Comment:
Is this the juju secret id or the vault secret id? Those two can easily be confused, I'd recomment being explicit here